typescript/cli/src/validator/preFlightCheck.ts

```diff
@@ -0,0 +1,136 @@
+  const merkleTreeHook = MerkleTreeHook__factory.connect(
+    addresses[chain].merkleTreeHook,
+    multiProvider.getProvider(chain),
+  );
+
+  let merkleTreeLatestCheckpointIndex: number | undefined;
+  try {
+    const [_, latestCheckpointIndex] = await merkleTreeHook[
+      'latestCheckpoint()'
+    ]();
+
+    merkleTreeLatestCheckpointIndex = latestCheckpointIndex;
+    logBlue(
+      `\nLatest check point index of incremental merkle tree: ${merkleTreeLatestCheckpointIndex}\n`,
+    );
+  } catch (err) {
+    warnYellow(
+      `❗️ Failed to fetch latest checkpoint index of merkleTreeHook on ${chain} \n`,
+    );
+  }
+
+  let announcedValidators;
+  try {
+    announcedValidators = await validatorAnnounce.getAnnouncedValidators();
+  } catch (err) {
+    errorRed(
+      `❌ Failed to fetch announced validators for ${chain}. Exiting.\n`,
+    );
+    process.exit(1);
+  }
+
+  const confirmedValidators = new Set<Address>();
+  const unconfirmedValidators = new Set<Address>();
+
+  for (const validator of validators) {
+    const matches = announcedValidators.filter((address) =>
+      eqAddress(address, validator),
+    );
+    if (matches.length === 0) {
+      unconfirmedValidators.add(validator);
+    } else {
+      confirmedValidators.add(validator);
+    }
+  }
+
+  const errorSet = new Set<string>();
+  errorSet.add('Validator pre flight check failed:');
+
+  if (unconfirmedValidators.size > 0) {
+    errorSet.add('Some validators have not been announced.');
+  }
+
+  let failedToReadError = false;
+  for (const validator of confirmedValidators) {
+    let storageLocations;
+    try {
+      storageLocations = await validatorAnnounce.getAnnouncedStorageLocations([
+        validator,
+      ]);
+      const s3StorageLocation = storageLocations[0][0];
+
+      const s3Validator = await S3Validator.fromStorageLocation(
+        s3StorageLocation,
+      );
+
+      const latestCheckpointIndex =
+        await s3Validator.getLatestCheckpointIndex();
+
+      logBlue(
+        `✅ Validator ${validator} announced\nstorage location: ${s3StorageLocation}\nlatest checkpoint index: ${latestCheckpointIndex}`,
+      );
+
+      // check is latestCheckpointIndex is within 1% of the merkleTreeLatestCheckpointIndex
+      if (merkleTreeLatestCheckpointIndex) {
+        const diff = Math.abs(
+          latestCheckpointIndex - merkleTreeLatestCheckpointIndex,
+        );
+        if (diff > merkleTreeLatestCheckpointIndex / 100) {
+          errorRed(
+            `❌ Validator is not signing the latest available checkpoint\n\n`,
+          );
+          errorSet.add(
+            `Some validators are not signing the latest available checkpoint\n`,
+          );
+        } else {
+          logBlue(
+            `✅ Validator is signing the latest available checkpoint\n\n`,
+          );
+        }
+      } else {
+        warnYellow(
+          `❗️ Cannot compare validator checkpoint signatures to latest check point \n`,
```

Comment:
```suggestion
          `❗️ No latest signed checkpoint found for validator\n`,
```
maybe to be more clear what happened

Reply:
We cant check whether the latest signed checkpoint for the validator is up to date because we were unable to read the latest checkpoint on `merkleTreeHook`, so I think `No latest signed checkpoint found for validator` also doesn't make it clear. 

`Latest checkpoint could not be read`?

Reply:
Ahh sorry I misread this. In that case then indicating why we can't compare and saying that the onchain latest checkpoint couldn't be read makes sense to me